ocs_ci/deployment/deployment.py

```diff
@@ -1172,6 +1175,94 @@ def deploy_acm_hub(self):
         """
         Handle ACM HUB deployment
         """
+        if config.ENV_DATA.get("acm_hub_downstream"):
+            self.deploy_acm_hub_downstream()
+        else:
+            self.deploy_acm_hub_upstream()
+
+    def deploy_acm_hub_downstream(self):
+        """
+        Handle AM HUB downstream deployment
+        """
+        logger.info("Cloning open-cluster-management deploy repository")
+        acm_hub_downstream_deploy_dir = os.path.join(
+            constants.EXTERNAL_DIR, "acm_hub_downstream_deploy"
+        )
+        clone_repo(
+            constants.ACM_HUB_DOWNSTREAM_DEPLOY_REPO, acm_hub_downstream_deploy_dir
+        )
+
+        logger.info("Retrieving quay token")
+        docker_config = load_auth_config().get("quay", {}).get("cli_password", {})
+        pw = base64.b64decode(docker_config)
+        pw = pw.decode().replace("quay.io", "quay.io:443").encode()
+        quay_token = base64.b64encode(pw).decode()
+
+        kubeconfig_location = os.path.join(self.cluster_path, "auth", "kubeconfig")
+
+        logger.info("Setting env vars")
+        env_vars = {
+            "QUAY_TOKEN": quay_token,
+            "COMPOSITE_BUNDLE": "true",
+            "CUSTOM_REGISTRY_REPO": "quay.io:443/acm-d",
+            "DOWNSTREAM": "true",
+            "DEBUG": "true",
+            "KUBECONFIG": kubeconfig_location,
+        }
+        for key, value in env_vars.items():
+            if value:
+                os.environ[key] = value
+
+        logger.info("Writing pull-secret")
+        _templating = templating.Templating(
+            os.path.join(constants.TEMPLATE_DIR, "acm-deployment")
+        )
+        template_data = {"docker_config": docker_config}
+        data = _templating.render_template(
+            constants.ACM_HUB_DOWNSTREAM_PULL_SECRET_TEMPLATE,
+            template_data,
+        )
+        pull_secret_path = os.path.join(
+            acm_hub_downstream_deploy_dir, "prereqs", "pull-secret.yaml"
+        )
+        with open(pull_secret_path, "w") as f:
+            f.write(data)
+
+        logger.info("Creating ImageContentSourcePolicy")
+        run_cmd(f"oc create -f {constants.ACM_HUB_DOWNSTREAM_ICSP_YAML}")
+
+        logger.info("Writing tag data to snapshot.ver")
+        # TODO: pull from config once this is set in OCP configs
+        tag_data = "2.4.3-DOWNSTREAM-2022-04-02-03-50-40"
+        with open(
+            os.path.join(acm_hub_downstream_deploy_dir, "snapshot.ver"), "w"
+        ) as f:
+            f.write(tag_data)
+
+        ci_dir = os.getcwd()
+        os.chdir(acm_hub_downstream_deploy_dir)
+        logger.info("Running open-cluster-management deploy")
+        cmd = [os.path.join(acm_hub_downstream_deploy_dir, "start.sh"), "--silent"]
+        logger.info("Running cmd: %s", " ".join(cmd))
+        proc = Popen(
+            cmd,
+            stdout=PIPE,
+            stderr=PIPE,
```

Comment:
I know that env should be taken from current process. But if you see still some issues that those env variable are not propagated well you can try to explicitly pass it here via env argument.

Reply:
I tried passing `env` explicitly with no change to the behavior of the script. As its more work to pull all of the base env vars and add them to the dict that is passed to the command, I opted to stick with the default behavior.